src/logicTemplates/split.js

```diff
@@ -66,58 +65,61 @@ class Split {
-     * @param {int} firstRound: the first round on which the transaction group will be valid
-     * @param {int} lastRound: the last round on which the transaction group will be valid
-     * @param {int} fee: the fee to pay in microAlgos
-     * @param {string} genesisHash: the b64-encoded genesis hash indicating the network for this transaction
-     * @param {boolean} precise, optional, precise treats the case where amount is not perfectly divisible based on the ratio.
-     *  When set to False, the amount will be divided as close as possible but one address will get
-     *  slightly more. When True, an error will be raised.
-     * @returns {Uint8Array}
-     */
-    getSendFundsTransaction(amount, firstRound, lastRound, fee, genesisHash, precise=true) {
-        let amountForReceiverOne = 0;
-        // reduce fractions
-        var gcdFn = function(a, b) {
-            if ((typeof a !== 'number') || (typeof b !== 'number')) throw "gcd operates only on positive integers";
-            if (!b) {
-                return a;
-            }
-            return gcdFn(b, a % b);
-        };
-        let gcd = gcdFn(this.ratn, this.ratd);
-        let ratn = Math.floor(this.ratn / gcd);
-        let ratd = Math.floor(this.ratd / gcd);
-        if (amount % ratd === 0) {
-            amountForReceiverOne = Math.floor(amount * ratn / ratd);
-        } else if (precise) {
-            throw Error("precise splitting requested but amount and contract ratio cannot be split precisely");
-        } else {
-            amountForReceiverOne = Math.round(amount * ratn / ratd);
+}
+/**
+ * returns a group transactions array which transfer funds according to the contract's ratio
+ * @param {Uint8Array} contract: bytes representing the contract in question
+ * @param {int} amount: the amount to be transferred
+ * @param {int} firstRound: the first round on which the transaction group will be valid
+ * @param {int} lastRound: the last round on which the transaction group will be valid
+ * @param {int} fee: the fee to pay in microAlgos
+ * @param {string} genesisHash: the b64-encoded genesis hash indicating the network for this transaction
+ * @returns {Uint8Array}
+ */
+function getSplitFundsTransaction(contract, amount, firstRound, lastRound, fee, genesisHash) {
+    let programOutputs = logic.readProgram(contract, undefined);
+    let ints = programOutputs[0];
+    let byteArrays = programOutputs[1];
+    let rat2 = ints[6];
+    let rat1 = ints[5];
+    let amountForReceiverOne = 0;
+    // reduce fractions
+    let gcdFn = function(a, b) {
+        if ((typeof a !== 'number') || (typeof b !== 'number')) throw "gcd operates only on positive integers";
+        if (!b) {
+            return a;
         }
-        let amountForReceiverTwo = amount - amountForReceiverOne;
-
-        let from = this.address;
+        return gcdFn(b, a % b);
+    };
+    let gcd = gcdFn(rat2, rat1);
+    rat2 = Math.floor(rat2 / gcd);
```

Comment:
Is it possible for this to round to the wrong number? (ex. something like 15/3 = 4.99999 and that gets rounded down to 4).